facts/get_facts.sh

```diff
@@ -74,15 +74,29 @@ case "${osfamily}" in
     done
     yum remove -y puppet7-release
   fi
+  wget "http://yum.puppetlabs.com/puppet8-release-el-${operatingsystemmajrelease}.noarch.rpm" -O /tmp/puppet8-release.rpm
+  if test -f /tmp/puppet8-release.rpm; then
+    rpm -ivh /tmp/puppet8-release.rpm
+    for puppet_agent_version in 8.0.0; do
+      if yum install -y puppet-agent-${puppet_agent_version}; then
+        output_file="/vagrant/$(facter --version | cut -d. -f1,2)/$(facter os.name | tr '[:upper:]' '[:lower:]')-$(facter os.release.major)-$(facter os.hardware).facts"
+        mkdir -p $(dirname ${output_file})
+        facter --show-legacy -p -j | tee ${output_file}
+      fi
+    done
+    yum remove -y puppet8-release
+  fi
   ;;
 'Debian')
-  apt_install curl
+  apt_install curl file
   curl "https://apt.puppetlabs.com/puppet6-release-${lsbdistcodename}.deb" -o /tmp/puppet6-release.deb
   # apt.puppetlabs.com returns an html document if the requested deb doesn't exist and /tmp/puppet6-release.deb will be an html doc
   if test "$?" -eq 0 -a -f /tmp/puppet6-release.deb && [[ "$(file -b /tmp/puppet6-release.deb)" =~ "Debian binary package".* ]] ; then
     dpkg --install /tmp/puppet6-release.deb
     apt-get update
-    for puppet_agent_version in 6.2 6.4 6.6; do
+    # as of 2023-04-27 older 6.x series aren't available (i.e. facter 3.12)
+    # the trailing dot is so that 6.4.* is the match so we get the last in the 6.4 series and not 6.40
+    for puppet_agent_version in 6.4. 6.28.; do
```

Comment:
given that Puppet 6 is EoL I would be fine with dropping the whole logic after we merged this PR.